### PR TITLE
Issue 395

### PR DESCRIPTION
**Pull request by nicolabertazzo**
fixed dspot maven plugin problem in filter and namesOfTestCases parameter.

### CODE REVIEW

File: dspot-maven/pom.xml [hunk 1]
```diff
@@ -38,6 +39,14 @@
 					</execution>
 				</executions>
 			</plugin>
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-surefire-plugin</artifactId>
+				<version>2.12.4</version>
+				<configuration>
+					<skipTests>true</skipTests>
```
Comment (danglotb) [inline dspot-maven/pom.xml]: Why are you skipping the test?

File: dspot-maven/pom.xml [hunk 2]
```diff
@@ -48,10 +57,17 @@
 			<artifactId>dspot</artifactId>
 			<version>${project.version}</version>
 		</dependency>
+
+		<dependency>
+			<groupId>eu.stamp-project</groupId>
+			<artifactId>dspot</artifactId>
+			<version>${project.version}</version>
+		</dependency>
```
Comment (danglotb) [inline dspot-maven/pom.xml]: Duplicated dependency to DSpot, right?

File: dspot/src/main/java/fr/inria/stamp/Main.java [hunk 3]
```diff
@@ -17,104 +17,141 @@
...
+		LOGGER.info("coverageDir: " + inputConfiguration.getCoverageDir());
+		LOGGER.info("outputDirectory: " + inputConfiguration.getOutputDirectory());
+		LOGGER.info("previousTransformationPath: " + inputConfiguration.getPreviousTransformationPath());
+		LOGGER.info("projectPath: " + inputConfiguration.getProjectPath());
+		LOGGER.info("relativeSourceCodeDir: " + inputConfiguration.getRelativeSourceCodeDir());
+		LOGGER.info("relativeTestSourceCodeDir: " + inputConfiguration.getRelativeTestSourceCodeDir());
+		LOGGER.info("RootPath: " + inputConfiguration.getRootPath());
+		LOGGER.info("TempDir: " + inputConfiguration.getTempDir());
+		LOGGER.info("ValidationErrors: " + inputConfiguration.getValidationErrors());
+	}
```
Comment (danglotb) [inline dspot/src/main/java/fr/inria/stamp/Main.java]: I kind of disagree with this large print... Quite useful for debugging I agree, but It appears to me as spam for users. Anyway, I would rather put this display code in a `toString()` methods of both objects `Configuration` and `InputConfiguration`, what do you think?

Comment (nicolabertazzo) [reply dspot/src/main/java/fr/inria/stamp/Main.java]: Yes it's for debug issue. It was useful for me to compare the parameter passed from dspot.propertis file with the parameter passed from pom.xml.
I agree we can move on toString() and activate it only on verbose mode.

File: dspot-maven/pom.xml (hunk 2 above)
Comment (nicolabertazzo) [reply dspot-maven/pom.xml]: No it's not a duplicated dependency. Dspot Maven plugin needs the DSpot dependency.

File: dspot-maven/pom.xml (hunk 1 above)
Comment (nicolabertazzo) [reply dspot-maven/pom.xml]: I forgot to remove the DSpotMojoTest and remove the skip tests configuration.
I created DSpotMojoTest test only for debug issue to check the execution of dspot on test-projects.

File: dspot-maven/pom.xml (hunk 2 above)
Comment (danglotb) [reply dspot-maven/pom.xml]: I meant in the pom, there is exactly the same dependency just above, _i.e._ dependency to **DSpot**, isn't?

**Pull request merged**